### PR TITLE
Add optout reason to hook call and optouts search api endpoint

### PR DESCRIPTION
**Pull request by erikh360**
opt-out rate trigger also included

### CODE REVIEW

**Comment by erikh360** [comment]
@rudigiesler please review.

File: identities/views.py [hunk 1]
```diff
@@ -257,6 +258,23 @@ def perform_create(self, serializer):
...
+class OptOutSearchList(generics.ListAPIView):
+    permission_classes = (IsAuthenticated,)
+    serializer_class = OptOutSerializer
+
+    def get_queryset(self):
+        query_params = list(self.request.query_params.keys())
+        filter_criteria = {}
+
+        for filter in query_params:
+            filter_criteria[filter] = self.request.query_params[filter]
```
Comment (rudigiesler) [inline identities/views.py]: What are we trying to do here? Are we not just recreating the dictionary exactly like it was before?

Comment (erikh360) [reply identities/views.py]: It is actually converting it from a QueryDict to Dict, if I use dict() to convert it I don't get the same results.

Comment (rudigiesler) [reply identities/views.py]: Can we not use the `.dict()` method? https://docs.djangoproject.com/en/1.10/ref/request-response/#django.http.QueryDict.dict

**Review by rudigiesler — APPROVED** [review]

**Pull request merged**